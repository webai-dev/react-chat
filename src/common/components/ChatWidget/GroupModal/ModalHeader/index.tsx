import React, { useContext } from "react";
import { Container, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import { CompanyContext } from "../../../../hooks/useCompanies";
import { ChatRoom } from '../../../../../types/types';

// Global styles
export const useStyles = makeStyles({
  root: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    alignItems: "center",
    height: "110px",
  },
  inputBox: {
    outline: "none",
    width: "512px",
    height: "58px",
    border: "1px solid #C7C7C7",
    paddingLeft: "16px",
    paddingRight: "16px",
    fontSize: "18px",
    fontFamily: "Roboto",
    fontWeight: 300,
    "&::placeholder": {
      fontSize: "18px",
      fontFamily: "Roboto",
      fontWeight: "300",
      color: "#43425D",
    },
  },
  title: {
    fontFamily: "Roboto",
    fontWeight: "bold",
    fontSize: "24px",
    color: "#43425D",
  },
});

interface ModalHeaderProps {
  name: string;
  setName: React.Dispatch<React.SetStateAction<string>>;
  chatRoom?: ChatRoom;
}

/**
 * ModalHeader component
 * @description ModalHeader component consists of Title and InputBox
 */
const ModalHeader: React.FC<ModalHeaderProps> = ({
  name = '',
  setName = () => undefined,
  chatRoom = null,
}) => {
  // Default styling
  const classes = useStyles();
  const { companies } = useContext(CompanyContext);
  const company = companies.find(company => chatRoom && company.id === chatRoom.companyId);

  return (
    <>
      <Container data-testid="modal-header" className={classes.root}>
        <Typography className={classes.title}>{company && company.name}</Typography>
        <input
          value={name}
          className={classes.inputBox}
          type="text"
          placeholder="Group chat name"
          onChange={(e) => setName(e.target.value)}
        />
      </Container>
    </>
  );
};

export default ModalHeader;
