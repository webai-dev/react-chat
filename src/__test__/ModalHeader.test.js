import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ModalHeader from "../common/components/ChatWidget/GroupModal/ModalHeader";
import renderer from "react-test-renderer";

it("renders without crashing", async () => {
  const { getByTestId } = render(<ModalHeader />);
  const modalHeader = getByTestId("modal-header");
  expect(modalHeader).toBeTruthy();
});

it("matches snapshot", async () => {
  const tree = renderer.create(<ModalHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});
