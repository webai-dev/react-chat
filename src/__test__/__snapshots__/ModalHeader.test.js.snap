// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiContainer-root makeStyles-root-4 MuiContainer-maxWidthLg"
  data-testid="modal-header"
>
  <p
    className="MuiTypography-root makeStyles-title-6 MuiTypography-body1"
  >
    Ternologic
  </p>
  <input
    className="makeStyles-inputBox-5"
    onChange={[Function]}
    placeholder="Group chat name"
    type="text"
    value=""
  />
</div>
`;
